llvm/lib/CAS/ObjectStore.cpp

```diff
@@ -56,6 +58,22 @@ void ReferenceBase::print(raw_ostream &OS, const ObjectRef &This) const {
   printReferenceBase(OS, "object-ref", InternalRef, ID);
 }
 
+void ObjectStore::loadAsync(
+    ObjectRef Ref,
+    unique_function<void(Expected<std::optional<ObjectHandle>>)> Callback) {
+  // The default implementation is synchronous.
+  Callback(load(Ref));
+}
+
+Expected<ObjectHandle> ObjectStore::loadExisting(ObjectRef Ref) {
+  std::optional<ObjectHandle> Handle;
+  if (Error E = load(Ref).moveInto(Handle))
+    return std::move(E);
+  if (!Handle)
+    return createStringError(errc::invalid_argument, "object missing from CAS");
```

Comment:
Could we add the ID string to the error?

Reply:
Done: https://github.com/apple/llvm-project/pull/6458/commits/f024c069ddec92ee6659811f42629f40204bfee0